src/opentimelineio/errorStatus.cpp

```diff
@@ -1,7 +1,12 @@
 #include "opentimelineio/errorStatus.h"
 
 namespace opentimelineio { namespace OPENTIMELINEIO_VERSION  {
-    
+
+bool ErrorStatus::is_ok(const ErrorStatus* es)
+{
+    return es && Outcome::OK == es->outcome;
```

Comment:
So the new behavior is to default to a nullptr ErrorStatus, and to treat the default as an error. I think that's a low touch change, as a step to getting rid of the bool operator first, and upgrading the error reporting pattern wholesale in the future.

Reply:
It's a fairly small change to the API though it will require a fair amount of work to make sure all the code using ```ErrorStatus``` can deal with null pointers.

I was looking at codecov to see how well the error cases were covered, and does this look right?
https://codecov.io/gh/PixarAnimationStudios/OpenTimelineIO/src/master/src/opentimelineio/composition.cpp#L165

In ```Composition::_index_of_child()``` it looks like the coverage shows the function never returns successfully in the tests, but the lines setting the error status are not recorded as hit (i.e., not green)? Actually the coverage in the rest of the files looks odd as well, the green numbers should show the number of times that line was hit, but they are all == 1?
https://docs.codecov.com/docs/viewing-source-code

Reply:
I'm not a codecov person, so I'll defer to someone else.

>  it will require a fair amount of work to make sure all the code using ErrorStatus can deal with null pointers.

Yep. The checks I put in were specifically ones I hit. We really need an audit of every function that takes an ErrorStatus.

Reply:
I've replaced the ErrorStatus operator with the ```is_ok``` function and checked all the uses of ```error_status``` to make sure the pointers are valid. It's a lot of changes but the tests are passing...